chips/atmega168-hal/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+            pins: {
+                sda: portc::PC4,
+                scl: portc::PC5,
+            },
+            registers: {
+                control: twcr {
+                    enable: twen,
+                    ack: twea,
+                    int: twint,
+                    start: twsta,
+                    stop: twsto,
+                },
+                status: twsr {
+                    prescaler: twps,
+                    status: tws,
+                },
+                bitrate: twbr,
+                data: twdr,
+            },
+        }
+    }
+}
+
+pub mod spi {
+    //! Implementation of the Rust Embedded-HAL SPI FullDuplex trait for AVR.
+    //!
+    //! The interface can be instantiated with the `new` method, and used directly
+    //! or passed into a driver.  Example usage:
+    //!
+    //! ```
+    //! pins.d10.into_output(&mut pins.ddr);// SS must be set to output mode
+    //! // create SPI interface
+    //! let mut spi = Spi::new(
+    //!     dp.SPI,// SPI peripheral
+    //!     pins.d13.into_output(&mut pins.ddr),// SCLK
+    //!     pins.d11.into_output(&mut pins.ddr),// MOSI output pin
+    //!     pins.d12.into_pull_up_input(&mut pins.ddr),// MISO input pin
+    //!     Settings::default(),
+    //! );
+    //!
+    //! // Send a byte
+    //! let sent = 0b10101010;
+    //! spi.send(sent).unwrap();
+    //! let response = spi.read().unwrap();
+    //! ```
+    //! In the example above, all of the settings are left at the default.  You can
+    //! also instantiate a Settings object with the other options available.
+
+    use crate::port::portb;
+    pub use avr_hal::spi::*;
+
+    avr_hal::impl_spi! {
+        pub struct Spi {
+            peripheral: crate::atmega168::SPI,
+            pins: {
+                sclk: portb::PB5,
+                mosi: portb::PB3,
+                miso: portb::PB4,
+            }
+        }
+    }
+}
+
+/// Serial interface using USART
+pub mod usart {
+    use crate::port::portd;
+    pub use avr_hal::serial::*;
+
+    crate::avr_hal::impl_usart! {
+        /// Serial interface based on ATmega168's USART0 peripheral
+        ///
+        /// Maximum baudrate seems to be 57600
+        pub struct Usart0 {
+            peripheral: crate::atmega168::USART0,
+            pins: {
+                rx: portd::PD0,
+                tx: portd::PD1,
+            },
+            registers: {
+                control_a: ucsr0a {
+                    data_empty: udre0,
+                    recv_complete: rxc0,
+                },
+                control_b: ucsr0b {
+                    tx_enable: txen0,
+                    rx_enable: rxen0,
+                },
+                control_c: ucsr0c {
+                    mode: umsel0,
+                    char_size: ucsz0,
+                    stop_bits: usbs0,
+                    parity: upm0,
+                },
+                baud: ubrr0,
+                data: udr0,
+            },
+        }
+    }
```

Comment:
This macro was changed a bit in commit 1e4941648a19 ("generic: serial: Refactor impl macro by using paste").  You'll have to adjust the macro invocation here similar to what I did for the other crates.

Reply:
Fixed in my local copy.